src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,153 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.NoSuchElementException;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
     @Override
     public void add(T value) {
+        Node<T> temp = tail;
+        Node<T> newNode = new Node<>(temp, value, null);
+        tail = newNode;
+        if (temp == null) {
+            head = newNode;
+        } else {
+            temp.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Index: " + index + ", is out of bound!");
+        }
+        if (index == size) {
+            add(value);
+        } else if (index == 0) {
```

Comment:
you can do like this 
```suggestion
        if (index == size) {
            add(value);
            return;
        }
        checkIndex(index)
        if (index == 0) {
```